qiskit_dynamics/solvers/solver_classes.py

```diff
@@ -180,40 +229,69 @@ def model(self) -> Union[HamiltonianModel, LindbladModel]:
 
     @property
     def signals(self) -> SignalList:
-        """The signals used in the solver.
-
-        These will be different from the signals in the model if a rotating wave approximation
-        was made.
-        """
+        """(Deprecated) The signals used in the solver."""
+        warnings.warn(
+            """The signals property is deprecated and will be removed in the next release.
+            Signals should be passed directly to the solve method.""",
+            DeprecationWarning,
+            stacklevel=2,
+        )
         return self._signals
 
     @signals.setter
     def signals(
         self, new_signals: Union[List[Signal], SignalList, Tuple[List[Signal]], Tuple[SignalList]]
     ):
-        """Set signals for the solver, and pass to the model."""
+        """(Deprecated) Set signals for the solver, and pass to the model."""
+        warnings.warn(
+            """The signals property is deprecated and will be removed in the next release.
+            Signals should be passed directly to the solve method.""",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
         self._signals = new_signals
         if self._rwa_signal_map is not None:
             new_signals = self._rwa_signal_map(new_signals)
         self.model.signals = new_signals
 
     def copy(self) -> "Solver":
-        """Return a copy of self."""
+        """(Deprecated) Return a copy of self."""
+        warnings.warn(
+            """The copy method is deprecated and will be removed in the next release.
+            This deprecation is associated with the deprecation of the signals property;
+            the copy method will no longer be needed once the signal property is removed.""",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
         return copy(self)
 
     def solve(
-        self, t_span: Array, y0: Union[Array, QuantumState, BaseOperator], **kwargs
-    ) -> OdeResult:
-        r"""Solve the dynamical problem.
-
-        Calls :func:`~qiskit_dynamics.solvers.solve_lmde`, and returns an `OdeResult`
-        object in the style of `scipy.integrate.solve_ivp`, with results
+        self,
+        t_span: Array,
+        y0: Union[Array, QuantumState, BaseOperator],
+        signals: Optional[Union[List[Signal], Tuple[List[Signal], List[Signal]]]] = None,
+        wrap_results: Optional[bool] = True,
```

Comment:
```suggestion
        wrap_results: bool = True,
```
You dont need optional if its always a bool